tests/FunctionalTestCase.php

```diff
@@ -51,6 +52,8 @@
 
 abstract class FunctionalTestCase extends TestCase
 {
+    private const ATLAS_TLD = '/\.(mongodb\.net|mongodb-dev\.net)/';
```

Comment:
UnifiedTestRunner has historically checked in a URI contained "mongodb.net" to detect an Atlas server. This pattern seems more robust, so I'd suggest making a similar change there. You can optionally address this in a separate PR, but I wouldn't object to doing it now.

This would require copying the `isAtlas()` method since UnifiedTestRunner does not extend this class.